src/features/signTx/logic_signTx.c

```diff
@@ -471,18 +436,7 @@ __attribute__((noinline)) static uint16_t finalize_parsing_helper(const txContex
             goto end;
         }
         if (G_called_from_swap) {
-            // Ensure the values are the same that the ones that have been previously validated
-            if (strcasecmp_workaround(strings.common.toAddress, displayBuffer) != 0) {
-                PRINTF("Error comparing destination addresses\n");
-                send_swap_error_with_string(APDU_RESPONSE_MODE_CHECK_FAILED,
-                                            SWAP_EC_ERROR_WRONG_DESTINATION,
-                                            APP_CODE_DEFAULT,
-                                            "%s != %s",
-                                            strings.common.toAddress,
-                                            displayBuffer);
-                // unreachable
-                os_sched_exit(0);
-            }
+            swap_check_destination(displayBuffer);
         } else {
             strlcpy(strings.common.toAddress, displayBuffer, sizeof(strings.common.toAddress));
         }
```

Comment:
We can simplify this and call strlcpy in both cases, and call swap_check_destination in the UI bypass with strings.common.toAddress as arg